pkg/operator/target_status.go

```diff
@@ -307,13 +320,14 @@ func patchPodMonitoringStatus(ctx context.Context, kubeClient client.Client, obj
 
 // updateTargetStatus populates the status object of each pod using the given
 // Prometheus targets.
-func updateTargetStatus(ctx context.Context, logger logr.Logger, kubeClient client.Client, targets []*prometheusv1.TargetsResult) error {
+func updateTargetStatus(ctx context.Context, logger logr.Logger, kubeClient client.Client, targets []*prometheusv1.TargetsResult, podMonitorings []monitoringv1.PodMonitoringCRD) error {
 	endpointMap, err := buildEndpointStatuses(targets)
 	if err != nil {
 		return err
 	}
 
 	var errs []error
+	hasEndpoints := map[string]bool{}
```

Comment:
In the spirit of operator memory / CPU challenges we see, can we remove this map entirely and modify `buildEndpointStatuses` to return the object name in key (so after `getObjectByScrapeJobKey`) instead of job key?

More readable and more efficienct?

..and if not, let's at least rename `hasEndpoints` to `withStatuses` or something. I was initially confused those are pod  monitorings with no endpoints configured. (: